src/backend/services/workflow_service.py

```
@@ -174,6 +183,16 @@ def run_agentic_workflow(natural_language_query: str, model_provider: str, model
         
         robot_code = '\n'.join(cleaned_lines).strip()
+        
+        # Step 3: Strip trailing JSON artifacts that may leak from LLM output
+        # LLM sometimes outputs {"code": "...robot code..."} and the closing "} leaks through
+        json_trailing_patterns = [
+            '"}',  # JSON closing brace with quote
+        ]
+        for pattern in json_trailing_patterns:
+            if robot_code.endswith(pattern):
+                robot_code = robot_code[:-len(pattern)].strip()
+                logging.info(f"✅ Stripped trailing JSON artifact: {pattern}")
```

Ruff (0.14.14)

`info()` call on root logger

(LOG015)